cobbler/modules/authorization/ownership.py

```diff
@@ -72,16 +80,26 @@ def __authorize_autoinst(api_handle, groups, user, autoinst) -> int:
     :param autoinst: The automatic installation in question.
     :return: ``1`` if the user is allowed and otherwise ``0``.
     """
-
-    lst = api_handle.find_profile(autoinst=autoinst, return_list=True)
-    lst.extend(api_handle.find_system(autoinst=autoinst, return_list=True))
+    lst: List[Union["Profile", "System"]] = []
+    my_profiles = api_handle.find_profile(autoinst=autoinst, return_list=True)
+    if my_profiles is None or not isinstance(my_profiles, list):
+        raise ValueError("list of profile was not a list")
+    lst.extend(my_profiles)
+    del my_profiles
```

Comment:
I will research that tomorrow.

Reply:
It was my assumption to have read in the [Google Python Style Guide](https://google.github.io/styleguide/pyguide.html) that this is a desired pattern. It seems I remembered this wrong.

However, the second reason for this I have is that the list of items in memory can potentially be very large. Thus I would like to free the memory. In this particular method, it seemed to be a non-risky change and thus I went for it since this PR is anyway for code quality.

Reply:
How big is the list? Do you have data to support the claim that this `del my_profiles` improves the memory footprint of Cobbler?

I'd be against starting optimizing the memory usage without any supporting data. Each explicit `del` statement is a surprise, they should at least come with a comment explaining why it's needed. We're not really optimizing memory usage in Cobbler, there are many low-risk improvements but if they don't change the overall footprint, I  don't think it should be a high priority to do them.

Reply:
Independent of your opinion if this is a nitpick or not I would love to get such a list of low-risk improvements in form of a GitHub discussion. The data you request can easily be generated with the test frameworks used in Cobbler. Generating this data I see as out of scope for this PR though.

Please advise if you would drop this from the PR or if you would just leave it.